internal/pkg/task/task.go

```diff
@@ -0,0 +1,112 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package task provides methods to run task in different configurations.
+package task
+
+import (
+	"errors"
+	"fmt"
+	"github.com/aws/copilot-cli/internal/pkg/aws/ec2"
+	"github.com/aws/copilot-cli/internal/pkg/aws/ecs"
+	"github.com/aws/copilot-cli/internal/pkg/aws/resourcegroups"
+	"github.com/aws/copilot-cli/internal/pkg/aws/session"
+)
+
+type vpcGetter interface {
+	GetSubnetIDs(filters ...ec2.Filter) ([]string, error)
+	GetSecurityGroups(filters ...ec2.Filter) ([]string, error)
+}
+
+type resourceGetter interface {
+	GetResourcesByTags(resourceType string, tags map[string]string) ([]*resourcegroups.Resource, error)
+}
+
+type defaultClusterGetter interface {
+	DefaultCluster() (string, error)
+}
+
+type taskRunner interface {
+	RunTask(input ecs.RunTaskInput) ([]string, error)
+}
+
+const (
+	startedBy           = "copilot-task"
+)
+
+var (
+	errNoSubnetFound     = errors.New("no subnets found")
+	fmtTaskFamilyName    = "copilot-%s"
+)
+
+// Task gets information required to run a task.
```

Comment:
What do you think of renaming the struct to `Runner`? so that it reads `task.Runner`.
```suggestion
// Runner can run an Amazon ECS task in either the default VPC or a Copilot VPC.
```

Reply:
I wonder if we should have an `"Env"` field in the struct so that instead of having two different methods `RunInDefaultVPC` and `RunInEnv` we can do:
```go
func (r *Runner) Run() ([]string, error) {
   if r.Env != "" {
        return r.runInEnv()
   }
   return r.runInDefaultVPC()
}
```

Reply:
Do you think same thing should be done for `RunInNetworkConfig`, which accepts the user provided subnets and security groups and run tasks in them? That is, having two fields called subnets and security groups, and make it a subroutine of `Run`

Reply:
I think that sounds good, or an alternative is to have three structs:
`EnvConfig`, `DefaultVPCConfig` and `NetworkConfig` all of them implement the same method signature: `Run() ([]string, error)`.

My goal was that in the `cli` pkg we shouldn't need to define 3 different interfaces to run a task, instead a single interface: `type runner interface { Run() ([]string, error }` that can be used for multiple different configurations.

Reply:
Yes that makes perfect sense!!